clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -175,7 +175,14 @@ static Address buildPointerWithAlignment(const Expr *E,
 
   // Unary &.
   if (const UnaryOperator *UO = dyn_cast<UnaryOperator>(E)) {
-    assert(0 && "not implemented");
+    // TODO(cir): maybe we should use cir.unary for pointers here instead.
```

Comment:
Good point! What I like about your pick for `cir.ptr_stride` is the fact that both `cir.ptr_stride` and `cir.ptr_diff` (and ptr_to_bool casts) isolate the usage of pointer on specific operations, perhaps making it easier for later pointer tracking passes to be implemented.